tests/Unit/MathTest.php

```diff
@@ -0,0 +1,175 @@
+{
+    public static function setUpBeforeClass(): void
+    {
+        require_once __DIR__ . '/../../src/math.php';
+    }
+
+    /**
+     * @covers euclidean_dist
+     * @dataProvider euclidianDistanceDataProvider
+     * @param array<float> $a
+     * @param array<float> $b
+     * @param float $dist
+     */
+    public function testEuclideanDist(array $a, array $b, float $dist): void
+    {
+        $this->assertEquals(round($dist, 6), round(euclidean_dist($a, $b), 6));
+    }
+
+    /**
+     * @return \Generator<array<mixed>>
+     */
+    public function euclidianDistanceDataProvider(): \Generator
+    {
+        /** @var array<string> $row */
+        foreach ($this->openCsv('euclidean_distances_2d.csv') as $row) {
+            list($x1, $y1, $x2, $y2, $dist) = array_map('floatval', $row);
+            yield [[$x1, $y1], [$x2, $y2], $dist];
+        }
+
+        /** @var array<string> $row */
+        foreach ($this->openCsv('euclidean_distances_3d.csv') as $row) {
+            list($x1, $y1, $z1, $x2, $y2, $z2, $dist) = array_map('floatval', $row);
+            yield [[$x1, $y1, $z1], [$x2, $y2, $z2], $dist];
+        }
+    }
+
+    /**
+     * @covers find_centroid
+     * @dataProvider centroidDataProvider
+     * @param array<float> $centroid
+     * @param array<float> ...$points
+     */
+    public function testFindCentroid(array $centroid, array ...$points): void
+    {
+        $this->assertEquals($centroid, find_centroid($points));
+    }
+
+    /**
+     * @return \Generator<array<array<float>>>
+     */
+    public function centroidDataProvider(): \Generator
+    {
+        /** @var array<string> $row */
+        foreach ($this->openCsv('centroids_2d.csv') as $row) {
+            list($x1, $y1, $x2, $y2, $x3, $y3, $x4, $y4, $cx, $cy) = array_map('floatval', $row);
+            yield [[$cx, $cy], [$x1, $y1], [$x2, $y2], [$x3, $y3], [$x4, $y4]];
+        }
+    }
+
+    /**
+     * @covers find_boundaries
+     * @dataProvider boundariesDataProvider
+     * @param array<float> $min
+     * @param array<float> $max
+     * @param array<float> ...$points
+     */
+    public function testFindBoundaries(array $min, array $max, array ...$points): void
+    {
+        $this->assertEquals([$min, $max], find_boundaries($points));
+    }
+
+    /**
+     * @return \Generator<array<array<float>>>
+     */
+    public function boundariesDataProvider(): \Generator
+    {
+        /** @var array<string> $row */
+        foreach ($this->openCsv('boundaries_2d.csv') as $row) {
+            list($x1, $y1, $x2, $y2, $x3, $y3, $x4, $y4, $x5, $y5, $ax, $ay, $bx, $by) = array_map('floatval', $row);
+            yield [[$ax, $ay], [$bx, $by], [$x1, $y1], [$x2, $y2], [$x3, $y3], [$x4, $y4], [$x5, $y5]];
+        }
+    }
+
+    /**
+     * @covers f_rand
+     * @dataProvider frandDataProvider
+     */
+    public function testFrand(float $min = 0, float $max = 1, float $nb = 1e5): void
+    {
+        assert($nb > 1e5);
```

Comment:
The expression "$nb > 1e5 " is always false.

Reply:
fixed.